apps/mobile/src/components/private-text.tsx

```diff
@@ -0,0 +1,36 @@
+import { Pressable } from 'react-native';
+
+import { useSettings } from '@/store/settings/settings';
+import { usePrivacyMode } from '@/store/settings/settings.read';
+
+import { Text, TextProps } from '@leather.io/ui/native';
+
+const defaultPrivateText = '***';
+
+interface PrivateTextProps {
+  color?: TextProps['color'];
+  privateText?: string;
+  text: string;
+  variant?: TextProps['variant'];
+}
+export function PrivateText({
```

Comment:
What do you think about this component not knowing anything about value it is wrapping? eg 

```
<PrivateValue>
  <Text>some text</Text>
</PrivateValue>
```
so that this component just either returns it's children or the private text?

Reply:
I thought about this, but the private text has to take on the styles of the 'children' ...so it didn't seem like the right approach. And, the private text itself isn't always the same. I'll look at it again to see if I missed something about doing the wrapper.

Reply:
Nice looks good! I like this encapsulation as the consumer needs to know a lot less about the implementation, and it has a smaller API surface area. Thanks for updating!